### PR TITLE
Iterate package

### PR DESCRIPTION
**Pull request by jalvz**
## Motivation/summary

This does a few things:

- Splits metrics data stream into app metrics, internal and profiles. This is following the indexing strategy proposal.
- Adds the dataset property to the manifest files so the right templates are created.
- Sort and dedup fields (within each yml file). Fields are now flattened. 
- Generates data stream pipelines out of our `definitions.json` file

~**Missing from this PR / WIP**: Pipelines are tied to data streams, we need to copy the ones that are used for other event types to the right data stream folder. Right now all are under `traces`. This shouldn't block the review.~

## Checklist

- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

I have considered changes for:
- [ ] documentation
- [ ] logging (add log lines, choose appropriate log selector, etc.)
- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))
- [ ] automated tests (add tests for the code changes, all [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally)
- [ ] telemetry
- [ ] Elasticsearch Service (https://cloud.elastic.co)
- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)
- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)

## How to test these changes

In short: run the package, install it, and check that all templates and pipelines installed are correct.

## Related issues

Part of #4004

### CODE REVIEW

**Review by axw — APPROVED** [review]

**Comment by axw** [comment]
Due to issues getting Elastic Agent + APM Server to run with 7.11.0 BC, and given that this isn't user facing yet, I'm testing with 8.0.0-SNAPSHOT.

- Installed package apm 0.1.0-dev.3
- Ran a program that sent a transaction, span, error, Go runtime metrics, custom metric
- Checked data streams:

image

- traces-apm-default contains the transaction and span
- logs-apm.error-default contains the error
- metrics-apm-default contains contains the Go runtime metrics and custom metric
- metrics-apm.internal-default contains breakdown metrics

That all seems good. I found a couple of issues.

The Data Streams list in the Fleet app does not list the "traces" data stream:

image

The "labels.foo" dynamic templates are missing from mappings:

image

Looking more closely at the package, I see that the fields.yml definitions for `labels` doesn't have all the bits from our overrides:

https://github.com/elastic/apm-server/blob/6d06e68ba8b02660c80a84fe3917e2361f72256c/apmpackage/apm/0.1.0/data_stream/traces/fields/ecs.yml#L130-L133
https://github.com/elastic/apm-server/blob/6d06e68ba8b02660c80a84fe3917e2361f72256c/model/transaction/_meta/fields.yml#L131-L141

**Comment by axw** [comment]
Opened https://github.com/elastic/apm-server/issues/4610 for the dynamic templates issue.

**Comment by axw** [comment]
The data stream list issue is due to this:

https://github.com/elastic/kibana/blob/94b02d97ed12c7d93081f88768def083b439473a/x-pack/plugins/fleet/server/routes/data_streams/handlers.ts#L12

I'll create a PR to fix that.

**Comment by axw** [comment]
For want of a better label, marking this "regression" given the labels issue.
